hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestFSPermissionChecker.java

```diff
@@ -446,4 +447,20 @@ private static INodeFile createINodeFile(INodeDirectory parent, String name,
     parent.addChild(inodeFile);
     return inodeFile;
   }
+
+  @Test
+  public void testCheckAccessControlEnforcerSlowness() throws Exception {
+    final long thresholdMs = 10;
+    final String message = FSPermissionChecker.runCheckPermission(() -> {
+      try {
+        Thread.sleep(20);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
```

Comment:
I suggest adding `Thread.currentThread().interrupt();` before throwing. It shouldn't matter much in practice, but JUnit runner threads have some strange behavior when interrupted status is not restored as expected.

Reply:
Sure. Thanks for pointing out the JUnit behavior.